rules/solid/tests/Rector/Variable/MoveVariableDeclarationNearReferenceRector/Fixture/variable_with_usage_in_multiple_levels.php.inc

```diff
@@ -0,0 +1,25 @@
+<?php
+
+function variableWithUsageInMultipleLevels()
+{
+    $var = 1;
+    if (mktime() === false) {
+        return $var;
+    }
+    echo $var;
```

Comment:
@TomasVotruba Is there some kind of helper in Rector to assist in finding the correct location for the assign statement in this situation? It looks like that would be a giant thing to write from scratch.

Reply:
I'm not aware of it. This Rule is the only place such service would be used in

Reply:
Definitely not an "easy pick" then :)
I'd have to get back to it later then, because this opens up a lot of potential use cases that we'd need to write for this Rector.

Reply:
Depends on the scope you go for :) No need to go so deep on 1st iteration. 

I'd be happy to merge it as it is without thins one case and then have a new PR.

Reply:
I'm not comfortable merging as is because that would cause the Rector to make incorrect changes and break the logic.

Reply:
Agreed. Broken code is not our goal.

That's why we go for opt-in approach, and skip such case completely. Then cover them in next PR.

Reply:
But we need to detect these cases to know that we need to skip them. If you unleash it as-is, in this particular case, rector would blindly move the assignment to the wrong location, causing the code to explode in one of the execution paths.

Reply:
That's true, but that's on purpose. It's the fastest way to discover such bugs, before it goes to general public. We release early and often, so bleeding edge testing can discover these hots spots ASAP.

Also, Rector is not a package in vendor, where breaking has different role.